### DISCUSSION TITLE
no support for function load?

### DISCUSSION THREAD

**Discussion by idanav111** [dragonflydb/dragonfly#2981] [Q&A]
Trying to load lua scripts as functions.

Generally when typing "function"  on the cli there is an error 
`(error) ERR Unknown subcommand or wrong number of arguments for 'HELP'. Try FUNCTION HELP.`

When trying to load the simplest lua function like: 

```
#!lua name=test

local function getUsers(keys, args)
    return "hello"
end

redis.register_function('getUsers', getUsers)
```

using the command
`redis-cli FUNCTION LOAD REPLACE "$(cat registerFunctions.lua)"`

The response is:
`(error) ERR wrong number of arguments for 'function' command`

When running the same on on redis server it works perfectly fine..

**Comment by chakaz** [comment]
Dragonfly doesn't support Redis functions at all, only `EVAL` scripts.
Can you modify your clients to use scripts instead?

**Reply by idanav111** [reply]
i can, its not as convenient as functions though.
Thanks!

**Reply by romange** [reply]
Can you explain why FUNCTION are more convenient?

**Reply by idanav111** [reply]
As far as i know, and i never used the EVAL script option, it will return a hash that i have to keep in order to use the lua script.
with functions, i simply call the function by its name like FCALL myFuncName 0 param1 param2
Using this way, i have a lua file like "registerFunctions.lua" with as many as functions that i need, 
and i just load them on server startup like
```
await client.scriptLoad(luaScript, {
 REPLACE: true
});
``` 

Also in this way, i dont have to worry about few servers that loads the same file or about updates on the lua scripts, because anyhow i will call the function by its name.

My alternative now with the EVAL scripts will be to save the hash somewhere and to have another functionality to update them if need.

**Reply by avifenesh** [reply]
They are persistent, spread across cluster, and reduce the payload size. Dragonfly still, and will, not supports it?
